src/Integration/State/StateManager.cs

```diff
@@ -127,8 +129,15 @@ private void OnIsBusyChanged(bool isBusy)
             this.IsBusyChanged?.Invoke(this, isBusy);
         }
 
-        private void OnBindingStateChanged()
+        private void OnBindingStateChanged(bool isBound)
```

Comment:
This method is called from both `ClearBoundProject` and `SetBoundProject`, however, these methods are not only called when the actual value really changes, but several times during the workflow. I introduced `isSolutionBound` to only raise `BindingStateChanged` when the binding status actually changes. But it's still not the case because the for example during a SQ binding "Update", the `ClearBoundProject` is being called. When the update finishes there's an appropriate `SetBoundProject` as well. So the end result is good, but still, I'm raising the `BindingStateChanged` unnecessarily.